src/wfs/src/main/java/org/geoserver/wfs/kvp/BaseFeatureKvpRequestReader.java

```diff
@@ -333,11 +342,12 @@ QName checkTypeName(QName qName, NamespaceSupport namespaces, EObject request) {
                     // namespace
                     namespaceURI = uri;
                 }
-            } else if (namespaces.getURI(prefix) != null) {
+            } else if ((namespaces.getURI(prefix) != null) && (qName.getNamespaceURI() != null)) {
```

Comment:
This change is unexpected for me, looks to me that GeoServer is trying to find in the provided namespaces a URI that matches the feature type prefix that was used in the request. For example:
```
typeNames=topp:states
namespaces=xmlns(topp,http://www.openplans.org/topp),xmlns(ns2,http://someserver.example.com/ns2)
```

... in this case he should match the ``topp`` prefix with the URI ``xmlns(topp,http://www.openplans.org/topp)``.